### PR TITLE
Fix missing is_deleted field on some tables

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #8547

### CODE REVIEW

**Comment by trasher** [comment]
Should fix for missing `is_deleted` fields. I've not yet found where the `is_template` is called, it's done explicitly on `glpi_computers` only in that class

**Review by orthagh — APPROVED** [review]

File: inc/stat.class.php [hunk 1]
```diff
@@ -748,9 +748,11 @@ static function constructEntryValues($itemtype, $type, $begin = "", $end = "", $
       $solved_status  = array_merge($closed_status, $item->getSolvedStatusArray());
 
       $criteria = [];
-      $WHERE           = [
-         "$table.is_deleted" => 0
-      ] + getEntitiesRestrictCriteria($table);
+      $WHERE = [];
+      if ($DB->fieldExists($table, 'is_deleted')) {
```
Comment (cedric-anne) [inline inc/stat.class.php:752]:
```suggestion
      if ($item->maybeDeleted()) {
```

File: inc/state.class.php [hunk 2]
```diff
@@ -119,16 +119,21 @@ static function showSummary() {
 
             } else {
                $table = getTableForItemType($itemtype);
+               $WHERE = [];
+               if ($DB->fieldExists($table, 'is_deleted')) {
```
Comment (cedric-anne) [inline inc/state.class.php:123]:
```suggestion
               if ($item->maybeDeleted()) {
```

File: inc/state.class.php [hunk 3]
```diff
@@ -119,16 +119,21 @@ static function showSummary() {
...
+                  $WHERE["$table.is_deleted"] = 0;
+               }
+               if ($DB->fieldExists($table, 'is_template')) {
```
Comment (cedric-anne) [inline inc/state.class.php:126]:
```suggestion
               if ($item->maybeTemplate()) {
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**